sessions/backend-update-delete/demo-end/components/JokeList/index.js

```diff
@@ -0,0 +1,30 @@
+import Button from "../Button";
+import List from "../List";
+import ListItem from "../ListItem";
+import useSWR from "swr";
+import { useRouter } from "next/router";
+
+export default function JokeList() {
+  const router = useRouter();
+  const { data } = useSWR("/api/jokes");
+
+  if (!data) {
+    return <h1>Loading...</h1>;
+  }
+
+  return (
+    <List>
+      {data.map((joke) => (
+        <ListItem key={joke._id}>
+          <Button
+            type="button"
+            width="100%"
+            onClick={() => router.push(`/${joke._id}`)}
+          >
+            {joke.joke}
```

Comment:
I am not sure but I think I remember that we wanted to use the Link component from Next.js here.

Reply:
@JessicaLoers let me know that we've also created an issue for this. If you're wondering why we're creating so many issues: it's a cascade of problems. This is the final stage of four consecutive sessions, so we have to address stuff that has been repeating and compounding once everything's done structurally. :)